okhttp/src/main/java/okhttp3/ConnectionSpec.java

```diff
@@ -83,11 +83,11 @@
   };
 
   /**
-   * A TLS 1.3 only Connection Spec for testing. This will be eventually be exposed
+   * A TLS 1.3 only Connection Spec. This will be eventually be exposed
    * as part of MODERN_TLS or folded into the default OkHttp client once published and
-   * available in JDK11.
+   * available in JDK11 or Conscrypt.
    */
-  static final ConnectionSpec TLS_13 = new Builder(true)
+  public static final ConnectionSpec TLS_13 = new Builder(true)
```

Comment:
Unsure about this in particular. Should we go more opinionated and just add TLS 1.3 to the modern set?

Reply:
Probably yes.  I'll re-add with the ciphers, since they are now formalised.

Reply:
There are 3 worrying results in the final set, particularly the google result.

```
Running tests using Jdk9Platform 11-ea+26
TLS1.3 only
https://enabled.tls13.com                java.net.ConnectException: Failed to connect to enabled.tls13.com/2400:cb00:2048:1:0:0:6810:7e22:443
https://www.howsmyssl.com/a/check        javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
https://tls13.cloudflare.com             java.net.ConnectException: Failed to connect to tls13.cloudflare.com/2400:cb00:2048:1:0:0:6813:c597:443
https://www.allizom.org/robots.txt       java.net.ConnectException: Failed to connect to www.allizom.org/2400:cb00:2048:1:0:0:6810:7e21:443
https://tls13.crypto.mozilla.org/        javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
https://tls.ctf.network/robots.txt       TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 404 169b
https://rustls.jbp.io/                   TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 200 416b
https://h2o.examp1e.net                  javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
https://mew.org/                         TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 137b
https://tls13.baishancloud.com/          javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
https://tls13.akamai.io/                 java.net.ConnectException: Failed to connect to tls13.akamai.io/2a02:26f0:5c:19f:0:0:0:5f37:443
https://swifttls.org/                    TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.0 200 222b
https://www.googleapis.com/robots.txt    java.net.ConnectException: Failed to connect to www.googleapis.com/2a00:1450:4009:801:0:0:0:200a:443
https://graph.facebook.com/robots.txt    TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 25b
https://api.twitter.com/robots.txt       javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
https://connect.squareup.com/robots.txt  javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure

TLS1.3 then fallback
https://enabled.tls13.com                TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 h2 200 140b
https://www.howsmyssl.com/a/check        TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 468b
https://tls13.cloudflare.com             TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 h2 521 4546b
https://www.allizom.org/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 h2 200 125b
https://tls13.crypto.mozilla.org/        javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
https://tls.ctf.network/robots.txt       TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 404 169b
https://rustls.jbp.io/                   TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 200 416b
https://h2o.examp1e.net                  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 h2 200 6209b
https://mew.org/                         TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 137b
https://tls13.baishancloud.com/          TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 2737b
https://tls13.akamai.io/                 TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 4324b
https://swifttls.org/                    TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.0 200 222b
https://www.googleapis.com/robots.txt    javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
https://graph.facebook.com/robots.txt    TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 25b
https://api.twitter.com/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 h2 200 155b
https://connect.squareup.com/robots.txt  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 0b

TLS1.3+TLS1.2
https://enabled.tls13.com                TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 h2 200 140b
https://www.howsmyssl.com/a/check        TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 518b
https://tls13.cloudflare.com             TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 h2 521 4546b
https://www.allizom.org/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 h2 200 125b
https://tls13.crypto.mozilla.org/        javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
https://tls.ctf.network/robots.txt       TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 404 169b
https://rustls.jbp.io/                   TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.1 200 416b
https://h2o.examp1e.net                  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 h2 200 6209b
https://mew.org/                         TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 137b
https://tls13.baishancloud.com/          javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
https://tls13.akamai.io/                 java.net.ConnectException: Failed to connect to tls13.akamai.io/2a02:26f0:5c:19f:0:0:0:5f37:443
https://swifttls.org/                    TLS_1_3 TLS_AES_128_GCM_SHA256 http/1.0 200 222b
https://www.googleapis.com/robots.txt    javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
https://graph.facebook.com/robots.txt    TLS_1_3 TLS_AES_128_GCM_SHA256 h2 200 25b
https://api.twitter.com/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 h2 200 155b
https://connect.squareup.com/robots.txt  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 0b
```

Reply:
Results with JDK 8

```
Running tests using Platform 25.162-b12
TLS1.3 only
https://enabled.tls13.com                java.net.ConnectException: Failed to connect to enabled.tls13.com/2400:cb00:2048:1:0:0:6810:7e22:443
https://www.howsmyssl.com/a/check        java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://tls13.cloudflare.com             java.net.ConnectException: Failed to connect to tls13.cloudflare.com/2400:cb00:2048:1:0:0:6813:c597:443
https://www.allizom.org/robots.txt       java.net.ConnectException: Failed to connect to www.allizom.org/2400:cb00:2048:1:0:0:6810:7e21:443
https://tls13.crypto.mozilla.org/        java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://tls.ctf.network/robots.txt       java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://rustls.jbp.io/                   java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://h2o.examp1e.net                  java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://mew.org/                         java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://tls13.baishancloud.com/          java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://tls13.akamai.io/                 java.net.ConnectException: Failed to connect to tls13.akamai.io/2a02:26f0:5c:1a2:0:0:0:5f37:443
https://swifttls.org/                    java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://www.googleapis.com/robots.txt    java.net.ConnectException: Failed to connect to www.googleapis.com/2a00:1450:4009:801:0:0:0:200a:443
https://graph.facebook.com/robots.txt    java.net.ConnectException: Failed to connect to graph.facebook.com/2a03:2880:f029:11:face:b00c:0:2:443
https://api.twitter.com/robots.txt       java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]
https://connect.squareup.com/robots.txt  java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_AES_128_CCM_SHA256, TLS_AES_256_CCM_8_SHA256], tlsVersions=[TLS_1_3], supportsTlsExtensions=true)], supported protocols=[TLSv1, TLSv1.1, TLSv1.2]

TLS1.3 then fallback
https://enabled.tls13.com                TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 140b
https://www.howsmyssl.com/a/check        TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 468b
https://tls13.cloudflare.com             TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 521 4546b
https://www.allizom.org/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 125b
https://tls13.crypto.mozilla.org/        javax.net.ssl.SSLException: Received fatal alert: protocol_version
https://tls.ctf.network/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 169b
https://rustls.jbp.io/                   TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 416b
https://h2o.examp1e.net                  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 6209b
https://mew.org/                         TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 137b
https://tls13.baishancloud.com/          TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 2737b
https://tls13.akamai.io/                 TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 4312b
https://swifttls.org/                    TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.0 200 237b
https://www.googleapis.com/robots.txt    TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 25b
https://graph.facebook.com/robots.txt    TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 25b
https://api.twitter.com/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 155b
https://connect.squareup.com/robots.txt  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 0b

TLS1.3+TLS1.2
https://enabled.tls13.com                TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 140b
https://www.howsmyssl.com/a/check        TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 468b
https://tls13.cloudflare.com             TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 521 4546b
https://www.allizom.org/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 125b
https://tls13.crypto.mozilla.org/        javax.net.ssl.SSLException: Received fatal alert: protocol_version
https://tls.ctf.network/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 169b
https://rustls.jbp.io/                   TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 416b
https://h2o.examp1e.net                  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 6209b
https://mew.org/                         TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 137b
https://tls13.baishancloud.com/          TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 2737b
https://tls13.akamai.io/                 TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.1 200 4312b
https://swifttls.org/                    TLS_1_2 TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 http/1.0 200 237b
https://www.googleapis.com/robots.txt    TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 25b
https://graph.facebook.com/robots.txt    TLS_1_2 TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 http/1.1 200 25b
https://api.twitter.com/robots.txt       TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 200 155b
https://connect.squareup.com/robots.txt  TLS_1_2 TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 http/1.1 404 0b
```